cmd/telegraf/cmd_win_service.go

```diff
@@ -0,0 +1,204 @@
+configuration(s) is specified the service will use the file in
+"C:\Program Files\Telegraf\telegraf.conf".
+
+To install Telegraf as a service use
+
+> telegraf service install
+
+In case you are planning to start multiple Telegraf instances as a service,
+you must use distrinctive service-names for each instance. To install two
+services with different configurations use
+
+> telegraf --config "C:\Program Files\Telegraf\telegraf-machine.conf" --service-name telegraf-machine service install
+> telegraf --config "C:\Program Files\Telegraf\telegraf-service.conf" --service-name telegraf-service service install
+`,
+					Flags: []cli.Flag{
+						&cli.StringFlag{
+							Name:  "display-name",
+							Value: "Telegraf Data Collector Service",
+							Usage: "service name as displayed in the service manager",
+						},
+						&cli.StringFlag{
+							Name:  "restart-delay",
+							Value: "5m",
+							Usage: "duration for delaying the service restart on failure",
+						},
+						&cli.BoolFlag{
+							Name:  "auto-restart",
+							Usage: "enable automatic service restart on failure",
+						},
+					},
+					Action: func(cCtx *cli.Context) error {
+						cfg := &serviceConfig{
+							displayName:  cCtx.String("display-name"),
+							restartDelay: cCtx.String("restart-delay"),
+							autoRestart:  cCtx.Bool("auto-restart"),
+
+							configs:    cCtx.StringSlice("config"),
+							configDirs: cCtx.StringSlice("config-directory"),
+						}
+						name := cCtx.String("service-name")
+						if err := installService(name, cfg); err != nil {
+							return err
+						}
+						fmt.Fprintf(outputBuffer, "Successfully installed service %q\n", name)
+						return nil
+					},
+				},
+				{
+					Name:  "uninstall",
+					Usage: "remove the Telegraf Windows service",
+					Description: `
+The 'uninstall' command removes the Telegraf service with the given name. To
+remove a service use
+
+> telegraf service uninstall
+
+In case you specified a custom service-name during install use
+
+> telegraf --service-name telegraf-machine service uninstall
+`,
+					Action: func(cCtx *cli.Context) error {
+						name := cCtx.String("service-name")
+						if err := uninstallService(name); err != nil {
+							return err
+						}
+						fmt.Fprintf(outputBuffer, "Successfully uninstalled service %q\n", name)
+						return nil
+					},
+				},
+				{
+					Name:  "start",
+					Usage: "start the Telegraf Windows service",
+					Description: `
+The 'start' command triggers the start of the Windows service with the given
+name. To start the service either use the Windows service manager or run
+
+> telegraf service start
+
+In case you specified a custom service-name during install use
+
+> telegraf --service-name telegraf-machine service start
+`,
+					Action: func(cCtx *cli.Context) error {
+						name := cCtx.String("service-name")
+						fmt.Fprintf(outputBuffer, "Starting service %s...\n", name)
```

Comment:
```suggestion
```

Reply:
@srebhan can we get rid of this one and call it good?

Reply:
Done.